notebooks/config-exp-cookbook.ipynb

```diff
@@ -0,0 +1,398 @@
+      "    namelist:\n",
+      "      update_values:\n",
+      "        config:\n",
+      "          cycle_day: !int '{{ cycle.strftime(''%d'') }}'\n",
+      "          cycle_hour: !int '{{ cycle.strftime(''%H'') }}'\n",
+      "          cycle_mon: !int '{{ cycle.strftime(''%m'') }}'\n",
+      "          convert_atm: true\n",
+      "          convert_nst: true\n",
+      "          convert_sfc: true\n",
+      "          data_dir_input_grid: /path/to/my/output/make_ics\n",
+      "          external_model: GFS\n",
+      "          input_type: gaussian_nemsio\n",
+      "          mosaic_file_target_grid: path/to/example_mosaic.halo.nc\n",
+      "          tg3_from_soil: false\n",
+      "          tracers:\n",
+      "          - sphum\n",
+      "          - liq_wat\n",
+      "          tracers_input:\n",
+      "          - spfh\n",
+      "          - clwmr\n",
+      "          varmap_file: /path/to/ufs-srweather-app/parm/ufs_utils/varmap_tables/GSDphys_var_map.txt\n",
+      "          vcoord_file_target_grid: path/to/global_hyblev.165.txt\n",
+      "    rundir: /path/to/my/output/make_ics\n",
+      "user:\n",
+      "  ACCOUNT: zrtrr\n",
+      "  MACHINE: hera\n",
+      "  PARMdir: /path/to/ufs-srweather-app/parm\n",
+      "workflow:\n",
+      "  EXPTDIR: /path/to/my/output\n"
+     ]
+    }
+   ],
+   "source": [
+    "experiment_config.dereference()\n",
+    "print(experiment_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5103465d-1a64-4b66-b9cb-09910633f8e1",
+   "metadata": {},
+   "source": [
+    "To catch potential errors in the `chgres_cube:` block of the config, the `uwtools` `chgres_cube` driver is used here to validate the block using the chgres-cube schema. The driver requires a `cycle` parameter with a datetime specified, so the current time is used here.\n",
+    "<!--cell 14-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "f9f0c0df-821e-492a-9669-3ac5e43e3151",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-11-18T13:34:51]     INFO Validating config against internal schema: chgres-cube\n",
+      "[2024-11-18T13:34:51]     INFO 0 UW schema-validation errors found\n",
+      "[2024-11-18T13:34:51]     INFO Validating config against internal schema: platform\n",
+      "[2024-11-18T13:34:51]     INFO 0 UW schema-validation errors found\n",
+      "[2024-11-18T13:34:51]     INFO 20241118 13Z chgres_cube valid schema: State: Ready\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "Asset(ref=None, ready=<function Assets.validate.<locals>.<lambda> at 0x7c3664345940>)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "driver = chgres_cube.ChgresCube(\n",
+    "    config=experiment_config['task_make_ics'],\n",
+    "    cycle=datetime.now()\n",
+    ")\n",
```

Comment:
```suggestion
    "driver = chgres_cube.ChgresCube(\n",
    "    config=experiment_config,\n",
    "    cycle=datetime.now()\n",
    ")\n",
```
I think maybe the `task_make_ics:` level in the config isn't adding anything here. Unless those YAML fixture files are used by something else (the Rocoto notebook?) that requires them, I think it would simplify the story here to remove that level in the configs. We use that additional level in practice in SRW because multiple `chgres_cube:` blocks are needed for different kinds of ICs, and nesting under unique keys supports that. But I'm not sure it demonstrates anything useful here. If we _do_ want to keep that level, I'd suggest that we use the `config=experiment_config, key_path=["task_make_ics"]` argument to the driver's constructor here instead of drilling down with `experiment_config['task_make_ics']`, as at least that introduces a useful new concept. But I would vote for simpler.

Reply:
I found that I got an error if the `chgres_cube:` block wasn't at the top level when validating, but I made the change here to use `key_path=["task_make_ics"]` instead.